tools/mdanalysis/ramachandran_plots.xml

```diff
@@ -0,0 +1,141 @@
+<tool id="mdanalysis_ramachandran_plot" name="Ramachandran Plots" version="0.1.3">
+    <description>Calculate and plot the distribution of two diheadrals in a trajectory</description>
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <expand macro="requirements" />
+    <command detect_errors="exit_code">
+<![CDATA[
+    python '$__tool_directory__/ramachandran_plots.py'
+        --idcd '$dcdin' 
+        --ipdb '$pdbin'
+        --isegid1  '$phi.segid1'
+        --iresid1  '$phi.resid1' 
+        --iname1   '$phi.name1'
+        --isegid2  '$phi.segid2'
+        --iresid2  '$phi.resid2' 
+        --iname2   '$phi.name2'
+        --isegid3  '$phi.segid3'
+        --iresid3  '$phi.resid3' 
+        --iname3   '$phi.name3'
+        --isegid4  '$phi.segid4'
+        --iresid4  '$phi.resid4' 
+        --iname4   '$phi.name4'
+        --isegid5  '$psi.segid1'
+        --iresid5  '$psi.resid1' 
+        --iname5   '$psi.name1'
+        --isegid6  '$psi.segid2'
+        --iresid6  '$psi.resid2' 
+        --iname6   '$psi.name2' 
+        --isegid7  '$psi.segid3' 
+        --iresid7  '$psi.resid3' 
+        --iname7   '$psi.name3' 
+        --isegid8  '$psi.segid4' 
+        --iresid8  '$psi.resid4' 
+        --iname8   '$psi.name4' 
+        --output   '$output'  
+        --oramachandran_plot '$ramachandran_plot'
+    2>&1
+]]></command>
+    <inputs>
+        <expand macro="analysis_inputs"/>
+        <section name="phi" title="Phi" expanded="False">
+          <param name="segid1"  type="text" value="HET" label="Segid of atom 1"/>
```

Comment:
the text fields could have some nice validators ... e.g. only letters, letters and numbers ...

Reply:
@bgruening user can use letters or numbers depending on the ID s used for segments they have used. Can you give me a example how to add validators?

Reply:
Sorry what I meant is https://docs.galaxyproject.org/en/latest/dev/schema.html#tool-inputs-param-sanitizer